api/app/api.py

```diff
@@ -19,7 +41,105 @@
     allow_headers=["*"]
 )
 
+class ForecastRequest(BaseModel):
+    site: PVSite
+    timestamp: Optional[str] = None
+    nwp_source: Optional[str] = "icon"
+    access_token: Optional[str] = None
+    enphase_system_id: Optional[str] = None
+
+class AuthUrlRequest(BaseModel):
+    full_auth_url: str
+
 @app.post("/forecast/")
-async def forecast(site:PVSite):
-    df =run_forecast(site)
-    return df.to_dict()
\ No newline at end of file
+async def forecast(request: ForecastRequest):
+    site = request.site
+    ts = request.timestamp if request.timestamp else pd.Timestamp.now(tz='UTC').isoformat()
+    nwp_source = request.nwp_source
+    access_token = request.access_token
+    enphase_system_id = request.enphase_system_id
+
+    try:
```

Comment:
please remove this try and except, fast api autoamtically gives a 500 if something goes wrong, so I dont think this is needed

Reply:
> please remove this try and except, fast api autoamtically gives a 500 if something goes wrong, so I dont think this is needed

alright will do

Reply:
> please remove this try and except, fast api autoamtically gives a 500 if something goes wrong, so I dont think this is needed

removed this try catch block